File: TaskManager-Angular/src/app/models/sign-up-view-model.ts
export class PersonFullName {
    firstName: string | any;
    lastName: string | any;

    constructor(){
        this.firstName = null;
        this.lastName = null;
    }
};

export class SignUpViewModel {

    personName : PersonFullName;
    email: string | any;
    mobile: string | any;
    dateOfBirth: string | any;
    password: string | any;
    confirPassword: string | any;
    gender: string | any;
    countryID: number | any;
    receiveNewsLetters: boolean | any;
    skills: any[];

    constructor() {

        this.personName = new PersonFullName();
        this.email = null;
        this.mobile = null;
        this.dateOfBirth = null;
        this.password = null;
        this.confirPassword = null;
        this.gender = null;
        this.countryID = null;
        this.receiveNewsLetters = false;
        this.skills = [];
    }
}

File: MvcTaskManger/MvcTaskManger/ViewModels/SignUpViewModel.cs
using MvcTaskManger.Models;
using System.Collections.Generic;

namespace MvcTaskManger.ViewModels
{
    public class SignUpViewModel
    {
        public PersonFullName PersonName { get; set; }

        public string Email { get; set; }

        public string Mobile { get; set; }

        public string DateOfBirth { get; set; }

        public string Password { get; set; }

        public string ConfirmPassword { get; set; }

        public string Gender { get; set; }

        public int CountryID { get; set; }

        public bool ReceiveNewLetters { get; set; }

        public List<Skill> Skills { get; set; }

    }


    public class PersonFullName
    {
        public string FirstName { get; set; }

        public string LastName { get; set; }
    }
}
